pkg/git/git_test.go

```diff
@@ -496,3 +498,47 @@ func TestTagsForBranchFailureWrongBranch(t *testing.T) {
 	require.NotNil(t, err)
 	require.Nil(t, result)
 }
+
+func TestCheckoutSuccess(t *testing.T) {
+	testRepo := newTestRepo(t)
+	defer testRepo.cleanup(t)
+
+	err := testRepo.sut.Checkout(Master, testRepo.testFileName)
+	require.Nil(t, err)
+}
+
+func TestCheckoutFailureWrongRevision(t *testing.T) {
+	testRepo := newTestRepo(t)
+	defer testRepo.cleanup(t)
+
+	err := testRepo.sut.Checkout("wrong")
+	require.NotNil(t, err)
+}
+
+func TestAddSuccess(t *testing.T) {
+	testRepo := newTestRepo(t)
+	defer testRepo.cleanup(t)
+
+	f, err := ioutil.TempFile(testRepo.sut.Dir(), "test")
+	require.Nil(t, err)
+	require.Nil(t, f.Close())
+
+	err = testRepo.sut.Add(filepath.Base(f.Name()))
+	require.Nil(t, err)
+}
+
+func TestAddFailureWrongPath(t *testing.T) {
+	testRepo := newTestRepo(t)
+	defer testRepo.cleanup(t)
+
+	err := testRepo.sut.Add("wrong")
+	require.NotNil(t, err)
+}
+
+func TestCommitSuccess(t *testing.T) {
+	testRepo := newTestRepo(t)
+	defer testRepo.cleanup(t)
+
+	err := testRepo.sut.Commit("msg")
+	require.Nil(t, err)
+}
```

Comment:
As far as I can tell, those don't really test much. If I'd change e.g. the commit method to
```go
func (r *Repo) Commit(_ string) error {
	return nil
}
```
the tests would still pass.

I suggest to either mock stuff your wrap (e.g. the "inner" `git.Repository`) and assert that certain methods have been called (with certain args, ...) (e.g. the `Worktree` or the `worktree.Commit` methods). Alternatively, as those tests are already integration-y tests anyway, assert on the expected end-state (check that a git object has actually been created)

Reply:
I see your point. Mocking would be a bit hard because `git.Repository` aka `inner` is a structure and not a public interface we can mock easily. I'll update the tests to test the end-state

Reply:
(Probably not for this PR, but we are heading in an interesting direction!)

> Mocking would be a bit hard because git.Repository aka inner is a structure and not a public interface

Right now, `inner` is a struct. But what if you decided it should be an interface (currently) with the methods `ResolveRevision`, `Tags`, `Remote`, `Worktree`, `CommitObject`, `Head`? 😎💥

Reply:
Yes this sounds good, I can follow up on that :+1: 
The next step after this PR would be to write some kind of integration test for the tool, probably side-by-side to the executable.

Reply:
My follow up on this: https://github.com/kubernetes/release/pull/1017